[GH-299] Sort views alphabetically after stripping leading emoji

Johennes opened the pull request:
#### Summary

This is a stopgap towards #299 that sorts views in the sidebar alphabetically but only after stripping any leading emoji. This prevents unintuitive results such as putting "📥 Inbox" before "🚀 Active".

*Before*

![Screenshot 2021-04-26 at 21 15 36](https://user-images.githubusercontent.com/1137962/116138019-99e2fe80-a6d4-11eb-800c-e0ee02f5a68c.png)

*After*

![Screenshot 2021-04-26 at 21 14 27](https://user-images.githubusercontent.com/1137962/116138021-9bacc200-a6d4-11eb-9489-a1fbfef19da3.png)

Still not sure if maybe this is too specialized a use case to warrant including this logic so happy to hear thoughts. 🙂 

#### Ticket Link

Relates to #299

mattermod commented:
Hello @Johennes,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Johennes commented on webapp/src/components/sidebar/sidebarBoardItem.tsx:
```diff
@@ -79,7 +79,19 @@ const SidebarBoardItem = React.memo((props: Props) => {
+            return 1
+        }
+        return 0
+    }).map((v) => v.view)
```
The linter forced me to spread this out over several lines like this (e.g. no nesting of ternaries allowed).

chenilim commented on webapp/src/components/sidebar/sidebarBoardItem.tsx:
```diff
@@ -79,7 +79,19 @@ const SidebarBoardItem = React.memo((props: Props) => {
+    const boardViews = views.filter((view) => view.parentId === board.id).map((v) => {
+        return {view: v, title: v.title.replace(/^\p{Emoji}*\s*/u, '')}
+    }).sort((v1, v2) => {
+        if (v1.title < v2.title) {
```
Thanks for the PR @Johennes! Looks great, but one minor thing here: we probably should use localeCompare here to handle international characters.

Johennes replied:
Ah, of course! 🤦 That actually also makes the code more terse, so win-win. 🎉

Johennes commented:
While changing to `localeCompare` I realized that the sorting logic might benefit from unit tests. I extracted a method for the sorting and added a few basic tests.

chenilim reviewed (approved):
Awesome. Thanks for the updates @Johennes! The code looks cleaner now and love the unit tests.

More a note for the future: I can see the need for "sort without leading emoji" as a common operation throughout the app.

Johennes commented:
> More a note for the future: I can see the need for "sort without leading emoji" as a common operation throughout the app.

Yeah, true. I think refactoring the sorting into a general utility function should be straightforward if we need it elsewhere in the future.
